pikelet/src/lang/core/semantics.rs

```diff
@@ -52,23 +53,11 @@ pub fn eval_term(
             Arc::new(Value::Sequence(value_entries))
         }
         Term::Ann(term, _) => eval_term(globals, universe_offset, values, term),
-        Term::RecordType(type_entries) => match type_entries.split_first() {
-            None => Arc::new(Value::RecordTypeEmpty),
-            Some(((entry_name, entry_type), rest_type)) => {
-                let entry_type = eval_term(globals, universe_offset, values, entry_type);
-                let rest_types = Closure::new(
-                    universe_offset,
-                    values.clone(),
-                    Arc::new(Term::RecordType(rest_type.to_owned())), // FIXME: expensive to_owned?
-                );
-
-                Arc::new(Value::RecordTypeExtend(
-                    entry_name.clone(),
-                    entry_type,
-                    rest_types,
-                ))
-            }
-        },
+        Term::RecordType(type_entries) => Arc::new(Value::RecordType(RecordTypeClosure::new(
+            universe_offset,
+            values.clone(),
+            type_entries.clone(),
```

Comment:
I like that we can just clone an `Arc` here, instead of doing the `<[_]>::to_owned` that we did previously (which would have allocated a new `Vec`.